pkg/tools/opa/policies/service_connection.rego

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2020 Cisco and/or its affiliates.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package nsm
+
+default service_connection = false
+
+service_connection {
+   conn_ids := [y | y = input.spiffe_id_connection_map[input.service_spiffe_id][_]]
+   count(input.path_segments) > 0
+   count(conn_ids) > 0
+   conn_ids == input.path_segments
+}
```

Comment:
Please note: the client is always previous segment. Please check only previous id.

Reply:
fixed